File: c++/647.PalindromicSubstrings/main.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

class Solution {
public:
    int countSubstrings(string s) {
        int count = 0;
        int n = s.length();
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < 2; j++) {
                int l = i;
                int r = i + j;
                while (l > -1 && r < n && s[l] == s[r]) {
                    count++;
                    l--;
                    r++;
                }
            }
        }
        return count;
    }
};

int main () {

    Solution sol;
    vector<pair<string, int> > tests = {
        {"abc", 3},
        {"aaa", 6},
        {"aba", 4},
        {"dnncbwoneinoplypwgbwktmvkoimcooyiwirgbxlcttgteqthcvyoueyftiwgwwxvxvg", 77}
    };

    bool failed = false;
    int testOnly = 0;
    int i = 0;
    for (auto test : tests) {
        if (testOnly != 0 && testOnly != i + 1) {
            continue;
        }
        string s = test.first;
        int expected = test.second;
        int actual = sol.countSubstrings(s);
        if (actual != expected) {
            failed = true;
            cout << "Test " << i + 1 << " Failed\n\tActual  :" << actual << "\n\tExpected: " << expected << "\n";
        }
    }
    if (!failed) {
        cout << "All Tests Passed.\n";
    }

}

File: Go/647.PalindromicSubstrings/main.go
package main

import (
	"fmt"
)

func countSubstrings(s string) int {
	count := 0
	n := len(s)
	for i := range n {
		for j := range 2 {
			l, r := i, i+j
			for l > -1 && r < n && s[l] == s[r] {
				count++
				l--
				r++
			}
		}
	}
	return count
}

func main() {
	tests := []struct {
		s        string
		expected int
	}{
		{"abc", 3},
		{"aaa", 6},
		{"aba", 4},
		{"dnncbwoneinoplypwgbwktmvkoimcooyiwirgbxlcttgteqthcvyoueyftiwgwwxvxvg", 77},
	}

	failed := false
	testOnly := 0
	for i, test := range tests {
		if testOnly != 0 && testOnly != i+1 {
			continue
		}
		actual := countSubstrings(test.s)
		if actual != test.expected {
			failed = true
			fmt.Printf("Test %v Failed\n\tActual  : %v \n\tExpected: %v\n", i+1, actual, test.expected)
		}
	}
	if !failed {
		fmt.Println("All Tests Passed.")
	}
}
